js/lex.go

```diff
@@ -33,6 +33,30 @@ const (
 	TemplateToken
 )
 
+// TokenState determines a state in which next token should be read
+type TokenState uint32
```

Comment:
Is there any difference? Probably Rust spoiled me, but now trying to use explicitly sized types everywhere.